script.js

```diff
@@ -1,48 +1,126 @@
-    if(gameState[clicked] === "") {
-        gameState[clicked] = currentPlayer;
-        return true;
-     }
-    return false;
+  if (gameState[clicked] === "") {
+    gameState[clicked] = currentPlayer;
+    return true;
+  }
+  return false;
 }
 
-function playGame() {
-    // this determines turn order
-    // this variable holds alternating player symbols
-    let currentPlayer = 'x';
-    const tiles = document.querySelectorAll('.tile');
-    startBtn.removeEventListener('click', playGame, false);
-    startBtn.style.opacity = 0.5;
-    tiles.forEach(function (tile) {
-        tile.addEventListener('click', () => {
-            //gets index value of tile from data attribute
-            let clicked = tile.dataset.index;
-            //assigns player marker based on currentPlayer value
-            //this is a boolean to regulate whether a player can click a tile.
-            let notYetClicked = alterGameState(gameState, clicked, currentPlayer);
-            //as long as tile has not been clicked,
-            //will add mark to gameState array
-            //and increment playerScore for next round
-            if(notYetClicked) {
-                addToScreen(tile, currentPlayer);
-                currentPlayer = decideTurn(currentPlayer);
-            }
-        })
+function clickEvent(event) {
+  let clicked = event.target.dataset.index;
+
+  let notYetClicked = alterGameState(gameState, clicked, currentPlayer);
+
+  if (notYetClicked) {
+    addToScreen(event.target, currentPlayer);
+    currentPlayer = decideTurn(currentPlayer);
+  }
+
+  event.target.removeEventListener("click", clickEvent);
 
+  let winner = checkWinner(gameState);
+
+  if (winner) {
+    showModal(winner)
+    startBtn.addEventListener("click", playGame);
+    startBtn.style.opacity = 1
+    tiles.forEach((tile) => {
+      tile.removeEventListener('click', clickEvent)
     })
+
+  }
+}
+
+function checkWinner(gameState) {
+  let winningCombinations = [
+    [0, 1, 2],
+    [3, 4, 5],
+    [6, 7, 8],
+    [0, 3, 6],
+    [1, 4, 7],
+    [2, 5, 8],
+    [0, 4, 8],
+    [2, 4, 6],
+  ];
+
+  let winMessage;
+
+  winningCombinations.forEach((set) => {
+    let indexA = set[0];
+    let indexB = set[1];
+    let indexC = set[2];
+
+    if (
+      gameState[indexA] === "x" &&
+      gameState[indexB] === "x" &&
+      gameState[indexC] === "x"
+    ) {
+      winMessage = "Player 1 wins!";
+    }
+
+    if (
+      gameState[indexA] === "o" &&
+      gameState[indexB] === "o" &&
+      gameState[indexC] === "o"
+    ) {
+      winMessage = "Player 2 wins!";
+    }
+
+  })
+
+  return winMessage;
 }
+
+function playGame() {
+  startBtn.removeEventListener('click', playGame, false);
+  startBtn.style.opacity = 0.5;
+
+  tiles.forEach((tile) => {
+    gameState = ["", "", "", "", "", "", "", "", ""]
+    tile.classList.remove('clickedX')
+    tile.classList.remove('clickedO')
+    tile.addEventListener('click', clickEvent)
+  })
+}
+
+function showModal(winner) {
```

Comment:
I love the modal, but maybe some wording to say something like 'hit start game to begin a new game) would help improve UX slightly